pkg/rpc/nodedialer/nodedialer_test.go

```diff
@@ -101,154 +105,254 @@ func TestDialNoBreaker(t *testing.T) {
-		}
-		return nil
+	testCases := []struct {
+		name        string
+		useDRPC     bool
+		dialFunc    func(ctx context.Context, nd *Dialer, nodeID roachpb.NodeID, class rpcbase.ConnectionClass) error
+		dialNoBreak func(ctx context.Context, nd *Dialer, nodeID roachpb.NodeID, class rpcbase.ConnectionClass) error
+	}{
+		{
+			name:    "gRPC",
+			useDRPC: false,
+			dialFunc: func(ctx context.Context, nd *Dialer, nodeID roachpb.NodeID, class rpcbase.ConnectionClass) error {
+				_, err := nd.Dial(ctx, nodeID, class)
+				return err
+			},
+			dialNoBreak: func(ctx context.Context, nd *Dialer, nodeID roachpb.NodeID, class rpcbase.ConnectionClass) error {
+				_, err := nd.DialNoBreaker(ctx, nodeID, class)
+				return err
+			},
+		},
+		{
+			name:    "DRPC",
+			useDRPC: true,
+			dialFunc: func(ctx context.Context, nd *Dialer, nodeID roachpb.NodeID, class rpcbase.ConnectionClass) error {
+				_, err := nd.DRPCDial(ctx, nodeID, class)
+				return err
+			},
+			dialNoBreak: func(ctx context.Context, nd *Dialer, nodeID roachpb.NodeID, class rpcbase.ConnectionClass) error {
+				_, err := nd.DRPCDialNoBreaker(ctx, nodeID, class)
+				return err
+			},
+		},
 	}
 
-	// When no connection exists, we expect ConnHealth to return ErrNotHeartbeated.
-	require.Equal(t, rpc.ErrNotHeartbeated, nd.ConnHealth(staticNodeID, rpcbase.DefaultClass))
-
-	// After dialing the node, ConnHealth should return nil.
-	_, err := nd.Dial(ctx, staticNodeID, rpcbase.DefaultClass)
-	require.NoError(t, err)
-	require.Eventually(t, func() bool {
-		return nd.ConnHealth(staticNodeID, rpcbase.DefaultClass) == nil
-	}, time.Second, 10*time.Millisecond)
-
-	// ConnHealth should still error for other node ID and class.
-	require.Error(t, nd.ConnHealth(9, rpcbase.DefaultClass))
-	require.Equal(t, rpc.ErrNotHeartbeated, nd.ConnHealth(staticNodeID, rpcbase.SystemClass))
-
-	// When the heartbeat errors, ConnHealth should eventually error too.
-	hb.setErr(errors.New("boom"))
-	require.Eventually(t, func() bool {
-		return nd.ConnHealth(staticNodeID, rpcbase.DefaultClass) != nil
-	}, time.Second, 10*time.Millisecond)
-
-	// When the heartbeat recovers, ConnHealth should too, assuming someone
-	// dials the node.
-	hb.setErr(nil)
-	{
-		// In practice this dial almost always immediately succeeds. However,
-		// because the previous connection was marked as unhealthy and might
-		// still just be about to be removed, the DialNoBreaker might pull
-		// the broken connection out of the pool and fail on it.
-		//
-		// See: https://github.com/cockroachdb/cockroach/issues/91798
-		require.Eventually(t, func() bool {
-			_, err := nd.DialNoBreaker(ctx, staticNodeID, rpcbase.DefaultClass)
-			return err == nil
-		}, 10*time.Second, time.Millisecond)
-		require.Eventually(t, func() bool {
-			return nd.ConnHealth(staticNodeID, rpcbase.DefaultClass) == nil
-		}, time.Second, 10*time.Millisecond)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
```

Comment:
Optional nit: We can use `testutils.RunTrueAndFalse` and since we have an `if useDRPC` statement, we can populate dialFunc and dialNoBreak there. It would reduce some code.